library/datasort.c

```diff
@@ -1106,14 +1110,15 @@ static struct datasort_chunk *datasort_merge(struct datasort_ctl *ds_ctl)
 		merged_chunk->index[total_count] = *dc;
 		merged_chunk->offset += dc->disk_size;
 		merged_chunk->count++;
+		merged_chunk->alive_count++;
 
 		// sync merged_chunk after each 1GB to control number of low priority dirty pages
 		if ((merged_chunk->offset - offset) > (1 << 30)) {
 			offset = merged_chunk->offset;
 			eblob_fdatasync(merged_chunk->fd);
 		}
 	}
-	assert(total_items == merged_chunk->count);
+	assert(total_items == merged_chunk->alive_count);
```

Comment:
It should also be equal to `merged_chunk->count` since we should merge only alive records.

Reply:
yes